### PR TITLE
👷Publish `aries-cloudagent-bbs`  Docker image

### LINKED ISSUE #3174: BBS enabled docker image overwrites normal image
In the [Enabling BBS Signatures Support](https://aca-py.org/latest/deploying/BBSSignatures/) documentation, it's insinuated that 2 docker images are built and published:
* `aries-cloudagent-python` - no BBS support, `linux/amd64` and `linux/arm64` supported
* `aries-cloudagent-bbs` - BBS support, `linux/amd64` only

However, in the Docker Publish action, the Extended (BBS enabled) image uses the same name and tag as the normal (BBS disabled) image, causing an overwrite as both use the same output of the same `meta` step:
```yaml
      - name: Setup Image Metadata
        id: meta
        uses: docker/metadata-action@v5
        with:
          images: |
            ghcr.io/${{ steps.info.outputs.repo-owner }}/aries-cloudagent-python
          tags: |
            type=raw,value=py${{ matrix.python-version }}-${{ inputs.tag || github.event.release.tag_name }}

      - name: Build and Push Image to ghcr.io
        uses: docker/build-push-action@v5
        with:
          push: true
          context: .
          file: docker/Dockerfile
          tags: ${{ steps.meta.outputs.tags }}
          labels: ${{ steps.meta.outputs.labels }}
          target: main
          build-args: |
            python_version=${{ matrix.python-version }}
            acapy_version=${{ inputs.tag || github.event.release.tag_name }}
          cache-from: type=local,src=/tmp/.buildx-cache
          cache-to: type=local,dest=/tmp/.buildx-cache-new,mode=max
          platforms: ${{ env.PLATFORMS }}

      - name: Build and Push extended Image to ghcr.io
        uses: docker/build-push-action@v5
        with:
          push: true
          context: .
          file: docker/Dockerfile
          tags: ${{ steps.meta.outputs.tags }} # This causes the previous (non-bbs) image to be overwritten
          labels: ${{ steps.meta.outputs.labels }}
          target: main
          build-args: |
            python_version=${{ matrix.python-version }}
            acapy_name=aries-cloudagent-bbs
            acapy_version=${{ inputs.tag || github.event.release.tag_name }}
            acapy_reqs=[askar,bbs,didcommv2]
          cache-from: type=local,src=/tmp/.buildx-cache
          cache-to: type=local,dest=/tmp/.buildx-cache-new,mode=max
          # Because of BBS, only linux/amd64 is supported for the extended image
          # https://github.com/hyperledger/aries-cloudagent-python/issues/2124#issuecomment-2293569659
          platforms: linux/amd64
```

I propose adding a dedicated `meta` step for the Extended (BBS enabled) image:
```yaml
      - name: Setup Image Metadata
        id: meta
        uses: docker/metadata-action@v5
        with:
          images: |
            ghcr.io/${{ steps.info.outputs.repo-owner }}/aries-cloudagent-python
          tags: |
            type=raw,value=py${{ matrix.python-version }}-${{ inputs.tag || github.event.release.tag_name }}

      - name: Build and Push Image to ghcr.io
        uses: docker/build-push-action@v5
        with:
          push: true
          context: .
          file: docker/Dockerfile
          tags: ${{ steps.meta.outputs.tags }}
          labels: ${{ steps.meta.outputs.labels }}
          target: main
          build-args: |
            python_version=${{ matrix.python-version }}
            acapy_version=${{ inputs.tag || github.event.release.tag_name }}
          cache-from: type=local,src=/tmp/.buildx-cache
          cache-to: type=local,dest=/tmp/.buildx-cache-new,mode=max
          platforms: ${{ env.PLATFORMS }}

      - name: Setup Image Metadata
        id: meta-bbs
        uses: docker/metadata-action@v5
        with:
          images: |
            ghcr.io/${{ steps.info.outputs.repo-owner }}/aries-cloudagent-bbs
          tags: |
            type=raw,value=py${{ matrix.python-version }}-${{ inputs.tag || github.event.release.tag_name }}

      - name: Build and Push extended Image to ghcr.io
        uses: docker/build-push-action@v5
        with:
          push: true
          context: .
          file: docker/Dockerfile
          tags: ${{ steps.meta-bbs.outputs.tags }}
          labels: ${{ steps.meta-bbs.outputs.labels }}
          target: main
          build-args: |
            python_version=${{ matrix.python-version }}
            acapy_name=aries-cloudagent-bbs
            acapy_version=${{ inputs.tag || github.event.release.tag_name }}
            acapy_reqs=[askar,bbs,didcommv2]
          cache-from: type=local,src=/tmp/.buildx-cache
          cache-to: type=local,dest=/tmp/.buildx-cache-new,mode=max
          # Because of BBS, only linux/amd64 is supported for the extended image
          # https://github.com/hyperledger/aries-cloudagent-python/issues/2124#issuecomment-2293569659
          platforms: linux/amd64
```
This will actually publish the missing `aries-cloudagent-bbs` image referenced in the [documentation](https://aca-py.org/latest/deploying/BBSSignatures/)

### PR DESCRIPTION
**Pull request by rblaine95**
* Resolves #3174
* Adds dedicated Docker Metadata step for the BBS enabled docker image
* Prevents the Extended (BBS enabled) image from overwriting the non-BBS image

### CODE REVIEW

**Review by amanji — COMMENTED** [review]
It LGTM but I'll defer to @esune or @WadeBarnes to give the thumbs up

**Review by jamshale — APPROVED** [review]
Surprised we missed this. Thanks for catching it. Looks good :+1:

**Comment by rblaine95** [comment]
@jamshale I should have caught it in #3171, thank you for being so quick to review 🙌

**Comment by jamshale** [comment]
Will merge after integration tests.

**Comment by swcurran** [comment]
So what is the impact of this going on after 1.0.0? Any concerns?

**Comment by jamshale** [comment]
I don't have any concerns. Doesn't affect any of the source code. We'll want to make sure the image is correct if it has been overridden by the bbs currently.

**Comment by jamshale** [comment]
That might involve deleting the current image and re-running the workflow. I'm not exactly sure.

**Comment by rblaine95** [comment]
@jamshale running the workflow manually and specifying the tag as `1.0.0` will override the existing `1.0.0` image without having to do another release

image

**Comment by jamshale** [comment]
Going to do that now.

**Comment by rblaine95** [comment]
Ah, that `403` error, I've seen that happen if it's the first time a package is being pushed.
Rerunning the workflow should resolve it.

**Comment by jamshale** [comment]
I've seen that too. Didn't know re-running it worked.

**Comment by rblaine95** [comment]
We (DIDx) had it happen a few times, especially when creating a large number of new packages in a single workflow run.

I think it's a hiccough with the Github API trying to provision new packages... some inconsistent backend state, something async that took just a little too long after creating the new package but before the permissions were configured in time 🤷

**Comment by jamshale** [comment]
Looks like it's published now :+1:

**Pull request merged**
